In [33]:
import pandas as pd
data=pd.read_csv('~/Library/CloudStorage/OneDrive-国立大学法人東海国立大学機構/Weekly_challenges/Data science and Analytics/Predict online behavior kaggle/Data/Raw/online_gaming_behavior_dataset.csv')
print(data.columns)
print(f"Number of (rows, columns) {data.shape}")
display(data.head(5))
print(data.info())
print(f"Missing values in each column: {data.isnull().sum()}")
print(f"duplicate rows: {data.duplicated().sum()}")

Index(['PlayerID', 'Age', 'Gender', 'Location', 'GameGenre', 'PlayTimeHours',
       'InGamePurchases', 'GameDifficulty', 'SessionsPerWeek',
       'AvgSessionDurationMinutes', 'PlayerLevel', 'AchievementsUnlocked',
       'EngagementLevel'],
      dtype='object')
Number of (rows, columns) (40034, 13)


PlayerID  Age  Gender Location GameGenre  PlayTimeHours  InGamePurchases  \
0      9000   43    Male    Other  Strategy      16.271119                0   
1      9001   29  Female      USA  Strategy       5.525961                0   
2      9002   22  Female      USA    Sports       8.223755                0   
3      9003   35    Male      USA    Action       5.265351                1   
4      9004   33    Male   Europe    Action      15.531945                0   

  GameDifficulty  SessionsPerWeek  AvgSessionDurationMinutes  PlayerLevel  \
0         Medium                6                        108           79   
1         Medium                5                        144           11   
2           Easy               16                        142           35   
3           Easy                9                         85           57   
4         Medium                2                        131           95   

   AchievementsUnlocked EngagementLevel  
0                    25          Medium  
1                    10          Medium  
2                    41            High  
3                    47          Medium  
4                    37          Medium

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40034 entries, 0 to 40033
Data columns (total 13 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   PlayerID                   40034 non-null  int64  
 1   Age                        40034 non-null  int64  
 2   Gender                     40034 non-null  object 
 3   Location                   40034 non-null  object 
 4   GameGenre                  40034 non-null  object 
 5   PlayTimeHours              40034 non-null  float64
 6   InGamePurchases            40034 non-null  int64  
 7   GameDifficulty             40034 non-null  object 
 8   SessionsPerWeek            40034 non-null  int64  
 9   AvgSessionDurationMinutes  40034 non-null  int64  
 10  PlayerLevel                40034 non-null  int64  
 11  AchievementsUnlocked       40034 non-null  int64  
 12  EngagementLevel            40034 non-null  object 
dtypes: float64(1), int64(7), object(5)
memory usag

In [32]:
#statistical summary
print("statitstical summary")
display(data.describe().T)

statitstical summary


count          mean           std          min  \
PlayerID                   40034.0  29016.500000  11556.964675  9000.000000   
Age                        40034.0     31.992531     10.043227    15.000000   
PlayTimeHours              40034.0     12.024365      6.914638     0.000115   
InGamePurchases            40034.0      0.200854      0.400644     0.000000   
SessionsPerWeek            40034.0      9.471774      5.763667     0.000000   
AvgSessionDurationMinutes  40034.0     94.792252     49.011375    10.000000   
PlayerLevel                40034.0     49.655568     28.588379     1.000000   
AchievementsUnlocked       40034.0     24.526477     14.430726     0.000000   

                                    25%           50%           75%  \
PlayerID                   19008.250000  29016.500000  39024.750000   
Age                           23.000000     32.000000     41.000000   
PlayTimeHours                  6.067501     12.008002     17.963831   
InGamePurchases                0.000000      0.000000      0.000000   
SessionsPerWeek                4.000000      9.000000     14.000000   
AvgSessionDurationMinutes     52.000000     95.000000    137.000000   
PlayerLevel                   25.000000     49.000000     74.000000   
AchievementsUnlocked          12.000000     25.000000     37.000000   

                                    max  
PlayerID                   49033.000000  
Age                           49.000000  
PlayTimeHours                 23.999592  
InGamePurchases                1.000000  
SessionsPerWeek               19.000000  
AvgSessionDurationMinutes    179.000000  
PlayerLevel                   99.000000  
AchievementsUnlocked          49.000000

In [38]:
#display number of unique values
print(f"number of unique values in each column:\n{data.nunique()}")

number of unique values in each column:
PlayerID                     40034
Age                             35
Gender                           2
Location                         4
GameGenre                        5
PlayTimeHours                40034
InGamePurchases                  2
GameDifficulty                   3
SessionsPerWeek                 20
AvgSessionDurationMinutes      170
PlayerLevel                     99
AchievementsUnlocked            50
EngagementLevel                  3
dtype: int64


In [46]:
#seperate numerical and categorical columns
numerical_columns=data.select_dtypes(include=['int64', 'float64']).columns.tolist()
non_numerical_columns=data.select_dtypes(include=['object']).columns.tolist()

print(f"Numerical columns are: \n {numerical_columns}")
print(f"Non numerical columns are: \n {non_numerical_columns}")
#display unique values for each categorical column
for col in non_numerical_columns:
    print(f"\n column: {col} \n unique values: {data[col].unique()}")

Numerical columns are: 
 ['PlayerID', 'Age', 'PlayTimeHours', 'InGamePurchases', 'SessionsPerWeek', 'AvgSessionDurationMinutes', 'PlayerLevel', 'AchievementsUnlocked']
Non numerical columns are: 
 ['Gender', 'Location', 'GameGenre', 'GameDifficulty', 'EngagementLevel']

 column: Gender 
 unique values: ['Male' 'Female']

 column: Location 
 unique values: ['Other' 'USA' 'Europe' 'Asia']

 column: GameGenre 
 unique values: ['Strategy' 'Sports' 'Action' 'RPG' 'Simulation']

 column: GameDifficulty 
 unique values: ['Medium' 'Easy' 'Hard']

 column: EngagementLevel 
 unique values: ['Medium' 'High' 'Low']


# Preleminary data exploration
- Data set size: 40034 rows and 13 columns: fairly large, providing robust base for statistical and machine learning analysis
- 8 numerical features and 5 categorical features: Gender, Location, GameGenre, GameDifficulty, and EngagementLevel, need encoding for use in predictive modeling.
- Target variable (engagement level) is categorized as High, Medium, Low
- No null values and no duplicates
- Numerical features:
* Age: 
1. Mean 32 years
2. Majority of players between 23 and 41 (interquartile)
3. min age: 15 and max age: 49
4. insight: players seem to be young to middle aged
* playTimeHours:
1. Mean 12 hours/session
2. Most players play between 6 and 18 hours
3. min: near zero and max: 24 hours
4. insight: players exhibit a wide range of play duration which could reflect casual vs hardcore gaming behaviors
* SessionPerWeek:
1. Mean: ~9 sessions per week
2. Players typically engage in 4–14 sessions weekly (Interquartile range)
3. Min: 0 and Max: 19 sessions; suggests heavy engagement for certain players.
4. Insights: The frequency of sessions per week is a key engagement indicator.
* AvgSessionDurationMinutes:
1. Mean: ~95 minutes per session.
2. Most sessions last 52–137 minutes.
3. Min: 10 and Max: 179 minutes per session.
4. Insights: Longer sessions may correlate with higher engagement and potential purchases.
* InGamePurchases:
1. Mean: ~20% of players make in-game purchases (1), while the rest do not (0).
2. Insights:A minority of players contribute to in-game monetization, likely forming a critical revenue segment.
* PlayerLevel:
1. Mean level: 50, with most players between level 25 and 74.
2. Min 1 and Max: Level 99.
3. Insights: Higher-level players could indicate seasoned gamers, likely influencing retention and engagement.
* AchievementsUnlocked:
1. Mean: 25 achievements unlocked.
2. Most players unlocked 12–37 achievements.
3. min 0 and max 49
4. Insights: Higher achievements unlocked may indicate stronger engagement or expertise.

# unique values analysis
PlayerID: 40,034 unique values (each player has a unique ID).
Age: 35 unique values, ranging from 15 to 49 years.
Gender: 2 unique values (Male, Female).
Location: 4 unique values (Other, USA, Europe, Asia).
GameGenre: 5 unique values (Strategy, Sports, Action, RPG, Simulation).
PlayTimeHours: 40,034 unique values (likely continuous data).
InGamePurchases: 2 unique values (binary: 0, 1).
GameDifficulty: 3 unique values (Medium, Easy, Hard).
SessionsPerWeek: 20 unique values, ranging from 0 to 19 sessions.
AvgSessionDurationMinutes: 170 unique values, representing session durations in minutes.
PlayerLevel: 99 unique values, ranging from 1 to 99.
AchievementsUnlocked: 50 unique values, ranging from 0 to 49.
EngagementLevel: 3 unique values (High, Medium, Low).
- Numerical Columns:
PlayerID, Age, PlayTimeHours, InGamePurchases, SessionsPerWeek, AvgSessionDurationMinutes, PlayerLevel, AchievementsUnlocked.
- Categorical Columns:
Gender, Location, GameGenre, GameDifficulty, EngagementLevel.